omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,298 @@
+    }
+
+    @Override
+    public IdentifierSource newDelegate() {
+        return new SequentialIdentifierGenerator();
+    }
+	
+    @Override
+    public Object create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		
+        Object savedIdentifierSource = null;
+        String identifierSourceType = validateInput(postBody.get("sourceType"));
+        ArrayList<String> exception = new ArrayList<String>();
+        String identifierTypeUuid = validateInput(postBody.get("identifierType"));
+        PatientIdentifierType patientIdentifierType = Context.getPatientService().getPatientIdentifierTypeByUuid(identifierTypeUuid);
+        String name = validateInput(postBody.get("name"));
+        String description = postBody.get("description").toString();
+
+        if (StringUtils.isBlank(identifierTypeUuid) || patientIdentifierType == null) {
+            exception.add("patient identifier type"); 
+        }
+        if (StringUtils.isBlank(name)) {
+            exception.add("name"); 
+        }
+        if (StringUtils.isBlank(description)) {
+            exception.add("description"); 
+        }
+
+        if(identifierSourceType.equals("SequentialIdentifierGenerator")) {
+
+            SequentialIdentifierGenerator identifierSource = new SequentialIdentifierGenerator();
+            String firstIdentifierBase = validateInput(postBody.get("firstIdentifierBase"));
+            String baseCharacterSet = validateInput(postBody.get("baseCharacterSet"));
+            String prefix = validateInput(postBody.get("prefix"));
+            String suffix = validateInput(postBody.get("suffix"));
+            String minLength = validateInput(postBody.get("minLength"));
+            String maxLength = validateInput(postBody.get("maxLength"));
+
+            if (StringUtils.isBlank(firstIdentifierBase)) { 
+                exception.add("first identifier base"); 
+            }
+            if (StringUtils.isBlank(baseCharacterSet)) { 
+                exception.add("base character set"); 
+            }
+            if (StringUtils.isBlank(baseCharacterSet)) { 
+                exception.add("base character set"); 
+            }
+            if (!StringUtils.isBlank(prefix)) { 
+                identifierSource.setPrefix(prefix); 
+            }
+            if (!StringUtils.isBlank(suffix)) { 
+                identifierSource.setSuffix(suffix); 
+            }
+            if (!StringUtils.isBlank(minLength)) {
+                if(StringUtils.isNumeric(minLength)) {
+                        identifierSource.setMinLength(Integer.parseInt(minLength));
+                }else {
+                        exception.add("min length");
+                }
+            }
+            if (!StringUtils.isBlank(maxLength)) {
+                if(StringUtils.isNumeric(maxLength)) {
+                        identifierSource.setMaxLength(Integer.parseInt(maxLength));
+                }else {
```

Comment:
Is max length required?

Reply:
No. Im checking if the value is set then it must be an integer. Otherwise it is skipped.

Reply:
So what does errors.add("max length") mean?

Reply:
I am adding "max length" to the error array if the user adds it in the request with invalid characters like: "12yy". It must only be an integer: "45"